### PR TITLE
Fix missing fields in acroform

### PR DESCRIPTION
**Pull request by jimmoffet**
Working with low level objects isn't super fun in pdf-lib, haha, but I got there. 

Ultimately, the offending pdf was poorly-authored (surprise) and some, but not all, widgets were added to the acroform catalog object. I'm guessing the author just gave up at the prospect of adding all those checkboxes to the acroform. Unfortunately, pdf-lib does not check for loose widgets that weren't added to the acroform object. `PDFForm` is essentially a 1:1 wrapper on acroform.

The solution was to iterate over all of the annotations with the widget type and add them to the doc's acroform manually. After that, we can grab the `PDFForm` and call `.getFields()` and get back those nice `PDFField` objects just like we did before.

This solution should generically capture any widgets missing from the acroform. I'm not sure if there will be a situation where there are widgets that the author intentionally kept outside of the acroform that we don't want in the acroform, but I can't think of one...

### CODE REVIEW

File: packages/forms/src/documents/pdf/extract.ts [hunk 1]
```diff
@@ -1,23 +1,105 @@
...
+  //   PDFName.of('Sound'),
+  //   PDFName.of('Movie'),
+  //   PDFName.of('Widget'),
+  //   PDFName.of('Screen'),
+  //   PDFName.of('PrinterMark'),
+  //   PDFName.of('TrapNet'),
+  //   PDFName.of('Watermark'),
+  //   PDFName.of('3D'),
+  //   PDFName.of('Redact'),
+  // ];
```
Comment (jimmoffet) [inline packages/forms/src/documents/pdf/extract.ts:53]: For reference, will stash somewhere

File: packages/forms/src/documents/pdf/extract.ts [hunk 2]
```diff
@@ -1,23 +1,105 @@
...
+    // console.log(widget);
+    newFields.push(widgetDictRef);
+  }
+
+  pdfDoc.catalog.set(
+    PDFName.of('AcroForm'),
+    pdfDoc.context.obj({
+      Fields: newFields,
+    })
+  );
```
Comment (jimmoffet) [inline packages/forms/src/documents/pdf/extract.ts:82]: This is where we add *all* of widgets to the acroform

File: packages/forms/src/documents/__tests__/fill-pdf.test.ts [hunk 3]
```diff
@@ -36,95 +36,108 @@ describe('PDF form filler', () => {
...
         label: 'Treasure',
+        maxLength: undefined,
         value: 'treasureField',
         required: false,
       },
-      Q0hBUkFDVEVSIElNQUdF: {
+      'Q0hBUkFDVEVSIElNQUdFLnVuZGVmaW5lZA==': {
         type: 'not-supported',
-        name: 'CHARACTER IMAGE',
+        name: 'CHARACTER IMAGE.undefined',
```
Comment (jimmoffet) [inline packages/forms/src/documents/__tests__/fill-pdf.test.ts:133]: edge case, looking into it

File: packages/forms/src/documents/pdf/extract.ts [hunk 4]
```diff
@@ -1,23 +1,57 @@
...
+    .enumerateIndirectObjects()
+    .map(([, obj]) => obj)
+    .filter(
+      obj =>
+        obj instanceof PDFDict &&
+        obj.get(PDFName.of('Type')) === PDFName.of('Annot') &&
+        obj.get(PDFName.of('Subtype')) === PDFName.of('Widget')
+    )
+    .map(obj => obj as PDFDict);
+};
```
Comment (jimmoffet) [inline packages/forms/src/documents/pdf/extract.ts:28]: Seems useful, so I'm exporting, but I don't actually feel strongly about it. It's ripped straight from pdf-lib, but seems to be only exposed on the acrofield object, which isn't useful in our case...

File: packages/forms/src/documents/pdf/extract.ts [hunk 5]
```diff
@@ -1,23 +1,57 @@
...
+  const pdfDoc = await PDFDocument.load(pdfBytes);
+  const widgets = await getWidgets(pdfDoc);
+
+  pdfDoc.catalog.set(
+    PDFName.of('AcroForm'),
+    pdfDoc.context.obj({
+      Fields: widgets.map(widget => pdfDoc.context.getObjectRef(widget)), // array of widget refs
+    })
+  );
+
```
Comment (jimmoffet) [inline packages/forms/src/documents/pdf/extract.ts:42]: This is where we add all of the widgets to the acroform, which will now get pulled into form via `pdfDoc.getForm()` below

File: packages/forms/src/documents/pdf/extract.ts [hunk 6]
```diff
@@ -1,23 +1,57 @@
...
+    .map(([, obj]) => obj)
+    .filter(
+      obj =>
+        obj instanceof PDFDict &&
+        obj.get(PDFName.of('Type')) === PDFName.of('Annot') &&
+        obj.get(PDFName.of('Subtype')) === PDFName.of('Widget')
+    )
+    .map(obj => obj as PDFDict);
+};
+
```
Comment (jimmoffet) [inline packages/forms/src/documents/pdf/extract.ts:29]: Seems useful, so I'm exporting, but I don't feel strongly about it. It's ripped straight from pdf-lib, but seems to be only exposed on the acrofield object, which isn't useful in our case...

**Review by danielnaab — APPROVED** [review]

File: packages/forms/src/documents/pdf/extract.ts (hunk 5 above)
Comment (danielnaab) [reply packages/forms/src/documents/pdf/extract.ts:42]: Interesting, so this is backfilling poorly annotated fields into the source PDF? That seems like an interesting approach. I like that it gives us a clear picture of the "error" of the source PDF, that we could perhaps inform the user about.

**Review by danielnaab — APPROVED** [review]
